src/libcore/panicking.rs

```diff
@@ -39,9 +39,15 @@
 use fmt;
 use panic::{Location, PanicInfo};
 
-#[cold] #[inline(never)] // this is the slow path, always
+#[cold]
+// inline(never) is required even in panic_immediate_abort mode, lest linker error
+#[inline(never)]
```

Comment:
It makes sense to me that `#[inline]` would cause a linker error, but does omitting the attribute entirely also cause a link error? (it's best to avoid `inline(never)` if we can in panic-immediate-abort mode)

Reply:
Yes, omitting `#[inline]` resolved the linker error. Updated the patch.